coala_quickstart/generation/SettingsFilling.py

```diff
@@ -32,38 +35,41 @@ def fill_section(section,
     """
     # Retrieve needed settings.
     prel_needed_settings = {}
+
     for bear in bears:
         needed = bear.get_non_optional_settings()
         for key in needed:
             if key in prel_needed_settings:
-                prel_needed_settings[key].append(bear.name)
+                prel_needed_settings[key]["bears"].append(bear.name)
             else:
-                prel_needed_settings[key] = [needed[key][0],
-                                             bear.name]
+                prel_needed_settings[key] = {
+                    "help_text": needed[key][0],
+                    "bears": [bear.name],
+                    "type": needed[key][1],
+                }
 
     # Strip away existent settings.
     needed_settings = {}
-    for setting, help_text in prel_needed_settings.items():
+    for setting, setting_info in prel_needed_settings.items():
         if setting not in section:
-            needed_settings[setting] = help_text
+            needed_settings[setting] = setting_info
 
     # Fill the settings with existing values if possible
     satisfied_settings = []
 
     for setting in needed_settings.keys():
         to_fill_values = list(autofill_value_if_possible(
-            setting, section.name, needed_settings[setting][1:],
+            setting, section.name, needed_settings[setting]["bears"],
```

Comment:
if you use "bears" a lot like in this, maybe you should define a global variable?